### ISSUE TITLE
Macro triggers clippy lint

### ISSUE THREAD

**Issue by purplesyringa** [danielhenrymantilla/fix_hidden_lifetime_bug.rs#14]
```rust
trait Trait {}
impl Trait for () {}

#[fix_hidden_lifetime_bug::fix_hidden_lifetime_bug]
fn test(_: &i32) -> impl Trait {}
```

This triggers the following clippy lint:
```
warning: the following explicit lifetimes could be elided: '_0
   --> src/lib.rs:543:12
    |
543 | fn test(_: &i32) -> impl Trait {}
    |            ^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
help: elide the lifetimes
    |
542 + 
543 ~ fn test(_: '_i32) -> impl Trait {}
    |
```

The macro effectively compiles to this:

```rust
trait Trait {}
impl Trait for () {}

trait Captures<'a> {}
impl<'a, T: ?Sized> Captures<'a> for T {}

fn test<'a>(_: &'a i32) -> impl Trait + Captures<'a> {}
```

but this would work too, without triggering clippy:

```rust
trait Trait {}
impl Trait for () {}

trait Captures<'a> {}
impl<'a, T: ?Sized> Captures<'a> for T {}

fn test(_: &i32) -> impl Trait + Captures<'_> {}
```

I'm thinking that this might be better codegen for elided lifetimes.

**Comment by danielhenrymantilla** [comment]
That's a style / _human-written_ code complexity lint; it shouldn't have been triggered for macro/machine-generated code, but that part is often disappointing.

I'll adjust the macro to `allow()` that lint, but will not make the macro logic ironically more _complex_ for the sake of soothing a subjective "complexity" lint 😄

**Comment by danielhenrymantilla** [comment]
Fixed as of v0.2.7: https://github.com/danielhenrymantilla/fix_hidden_lifetime_bug.rs/commit/fe1c5e865dd4c26e49687346e2dcac610c50d024

**Comment by purplesyringa** [comment]
Thanks. I'm wondering if this fix disables the clippy lint even for user code inside the function body?
